platforms/org.eclipse.mita.platform.xdk110/src/org/eclipse/mita/platform/xdk110/connectivity/MqttGenerator.xtend

```diff
@@ -316,20 +336,29 @@ class MqttGenerator extends AbstractSystemResourceGenerator {
 		.addHeader("MbedTLSAdapter.h", true)
 		.addHeader("HTTPRestClientSecurity.h", true)
 		.addHeader("time.h", true);
-		if(auth instanceof SumTypeRepr) {
-			if(auth.isLogin()) {
-				val username = auth.properties.get("username")?.code;
-				val password = auth.properties.get("password")?.code;
-				result.setPreamble('''
-
-                StringDescr_T username;
-                const char* usernameBuf = «username»;
-
-                StringDescr_T password;
-                const char* passwordBuf = «password»;
-				''').addHeader("Serval_StringDescr.h", true);
-			}
-		}
+		result.setPreamble('''
+		«IF(auth instanceof SumTypeRepr)»
```

Comment:
don't need the parentheses for the if statement

Reply:
right you are!